### PR TITLE
chore: cut 0.77.0

### PR DESCRIPTION
**Pull request by slashpai**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

Full packed release 🚀 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,28 @@
 ## Unreleased
 
-* [FEATURE] Add `-kubelet-endpointslice` argument to support `EndpointSlice` for the kubelet controller.
+## 0.77.0 / 2024-09-18
+
```
Comment (nicolastakashi) [inline CHANGELOG.md:4]: I think we can remove this extra line

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+> Along with this release we also introduce a new command line tool [poctl](https://github.com/prometheus-operator/poctl). It is designed specifically for managing Prometheus Operator Custom Resources.
+
+> This project is currently under active development and fully experimental, so expect breaking changes and rough edges. We encourage you to [try it out](https://github.com/prometheus-operator/poctl/releases/tag/v0.1.0) and provide the feedback.
+
+* [CHANGE] Add more API validation and changes in ScrapeConfig CRD. #6921 #6855
+* [CHANGE] Update API validations for ECSDConfig in ScrapeConfig CRD. #6744
+* [FEATURE] Add experimental Prometheus 3 support. #6940
```
Comment (nicolastakashi) [inline CHANGELOG.md:11]: I was wondering if also makes sense highlight Prometheus 3 is experimental and we're not supporting it, wdyt?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,28 @@
 ## Unreleased
 
-* [FEATURE] Add `-kubelet-endpointslice` argument to support `EndpointSlice` for the kubelet controller.
+## 0.77.0 / 2024-09-18
+
+> Along with this release we also introduce a new command line tool [poctl](https://github.com/prometheus-operator/poctl). It is designed specifically for managing Prometheus Operator Custom Resources.
+
```
Comment (simonpasquier) [inline CHANGELOG.md:6]: remove this line to avoid a breaking line?

File: CHANGELOG.md (hunk 1 above)
Comment (slashpai) [reply CHANGELOG.md:4]: we need blank line after heading for markdown

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,28 @@
...
+> This project is currently under active development and fully experimental, so expect breaking changes and rough edges. We encourage you to [try it out](https://github.com/prometheus-operator/poctl/releases/tag/v0.1.0) and provide the feedback.
+
+* [CHANGE] Add more API validation and changes in ScrapeConfig CRD. #6921 #6855
+* [CHANGE] Update API validations for ECSDConfig in ScrapeConfig CRD. #6744
```
Comment (simonpasquier) [inline CHANGELOG.md:9]: IMHO either we have a catch-all entry for ScrapeConfig CRD or we have one line per PR.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,28 @@
...
+* [FEATURE] Add experimental Prometheus 3 support. #6940
+* [FEATURE] Allow Prometheus Agent to update, delete, and handle changes to DaemonSet object. #6787
+* [FEATURE] Add support PodMonitor for Prometheus Agent in DaemonSet mode. #6879
```
Comment (simonpasquier) [inline CHANGELOG.md:13]:
```suggestion
* [FEATURE] Add support for `PodMonitor` to the `PrometheusAgent` CRD in DaemonSet mode. #6879
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,28 @@
...
+* [FEATURE] Add `-kubelet-endpointslice` argument to support `EndpointSlice` for the kubelet controller. #6882
+* [FEATURE] Reject PrometheusRule objects that generate > 512KiB configmap. #6606
```
Comment (simonpasquier) [inline CHANGELOG.md:15]:
```suggestion
* [CHANGE] Update the validating webhook to reject `PrometheusRule` objects that would generate configuration greater than the max ConfigMap size (512KiB). #6606
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,28 @@
...
+* [FEATURE] Add support for `OutOfOrderTimeWindow` field for PrometheusAgent. #6851
```
Comment (simonpasquier) [inline CHANGELOG.md:16]:
```suggestion
* [FEATURE] Add `outOfOrderTimeWindow` field to the `PrometheusAgent` CRD. #6851
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,28 @@
...
+* [CHANGE] Add more API validation and changes in ScrapeConfig CRD. #6921 #6855
+* [CHANGE] Update API validations for ECSDConfig in ScrapeConfig CRD. #6744
+* [FEATURE] Add experimental Prometheus 3 support. This will allow users to try new Prometheus `v3.0.0-beta.0`. #6940
+  * Feature requests in this version won't be supported till Prometheus 3 is stable but this would let users to try the beta version with Operator and report issues if any.
+* [FEATURE] Allow Prometheus Agent to update, delete, and handle changes to DaemonSet object. #6787
+* [FEATURE] Add support PodMonitor for Prometheus Agent in DaemonSet mode. #6879
+* [FEATURE] Add `-kubelet-endpointslice` argument to support `EndpointSlice` for the kubelet controller. #6882
+* [FEATURE] Reject PrometheusRule objects that generate > 512KiB configmap. #6606
+* [FEATURE] Add support for `OutOfOrderTimeWindow` field for PrometheusAgent. #6851
+* [FEATURE] Add support for `MatchFirstNetwork` field for DockerSD. #6849 #6894
```
Comment (simonpasquier) [inline CHANGELOG.md:17]:
```suggestion
* [FEATURE] Add `matchFirstNetwork` field to the `ScrapeConfig` CRD for the Docker Service Discovery. #6849 #6894
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,6 +1,28 @@
...
+* [FEATURE] Add support for new otlp receiver configuration. #6780
```
Comment (simonpasquier) [inline CHANGELOG.md:18]:
```suggestion
* [FEATURE] Add the `otlp` field to the `Prometheus` and `PrometheusAgent` CRDs. #6780
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,6 +1,28 @@
...
+* [FEATURE] Add support for file paths for certificate, key and client CA in web TLS. #6358
```
Comment (simonpasquier) [inline CHANGELOG.md:19]:
```suggestion
* [FEATURE] Add support for certificate, key and client CA file paths to the web TLS configuration. #6358
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,6 +1,28 @@
...
+* [FEATURE] Add new proxy fields into the AlertmanagerConfig CRD for the OAuth2 clients. #6922
```
Comment (simonpasquier) [inline CHANGELOG.md:20]:
```suggestion
* [FEATURE] Add `noProxy`, `proxyFromEnvironment` and `proxyConnectHeader` fields to the AlertmanagerConfig CRD for OAuth2 clients. #6922
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,6 +1,28 @@
...
+* [FEATURE] Add ProxyConfig fields into the AlertmanagerConfig CRD for the HTTP clients. #6864
```
Comment (simonpasquier) [inline CHANGELOG.md:21]:
```suggestion
* [FEATURE] Add `noProxy`, `proxyFromEnvironment` and `proxyConnectHeader` fields to the AlertmanagerConfig CRD for HTTP clients. #6864
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,6 +1,28 @@
...
+* [FEATURE] Add Ionos Service Discovery support to ScrapeConfig CRD. (#6732)
```
Comment (simonpasquier) [inline CHANGELOG.md:22]:
```suggestion
* [FEATURE] Add Ionos Service Discovery support to `ScrapeConfig` CRD. (#6732)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,6 +1,28 @@
...
+* [BUGFIX] Fix AlertmanagerConfig objects with invalid child routes definition to cause problems. #6931
```
Comment (simonpasquier) [inline CHANGELOG.md:23]: I added it to the CHANGELOG.md already but at a wrong place :facepalm: 
https://github.com/prometheus-operator/prometheus-operator/blob/c846674a17c6db9d1bd04d08300c354f960d909b/CHANGELOG.md?plain=1#L11

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,6 +1,28 @@
...
+* [BUGFIX] Fix avoid BADKEY error in logs. #6936
```
Comment (simonpasquier) [inline CHANGELOG.md:24]: We can omit this one because it wasn't part of an official release.

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,6 +1,28 @@
...
+* [BUGFIX] Allow proxy URL for all Prometheus versions. #6845
```
Comment (simonpasquier) [inline CHANGELOG.md:25]:
```suggestion
* [BUGFIX] Allow configuring proxy URL for all Prometheus versions. #6845
```

**Review by simonpasquier — COMMENTED** [review]
Wow, this is a huge release! We can also mention that we've switched the logging framework from gokit-log to slog which may lead to some differences in the log output.

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,6 +1,26 @@
 ## Unreleased
 
-* [FEATURE] Add `-kubelet-endpointslice` argument to support `EndpointSlice` for the kubelet controller.
+## 0.77.0 / 2024-09-18
+
+> Along with this release we also introduce a new command line tool [poctl](https://github.com/prometheus-operator/poctl). It is designed specifically for managing Prometheus Operator Custom Resources.
+> This project is currently under active development and fully experimental, so expect breaking changes and rough edges. We encourage you to [try it out](https://github.com/prometheus-operator/poctl/releases/tag/v0.1.0) and provide the feedback.
+
+* [CHANGE] Add more API validation and changes in `ScrapeConfig CRD`. #6921 #6855 #6744
```
Comment (simonpasquier) [inline CHANGELOG.md:8]:
```suggestion
* [CHANGE] Add more API validation and changes to the `ScrapeConfig` CRD. #6921 #6855 #6744
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,6 +1,26 @@
...
+* [CHANGE] Update the validating webhook to reject `PrometheusRule` objects that would generate configuration greater than the max ConfigMap size (512KiB). #6606
+* [FEATURE] Add experimental Prometheus 3 support. This will allow users to try new Prometheus `v3.0.0-beta.0`. #6940
```
Comment (simonpasquier) [inline CHANGELOG.md:10]:
```suggestion
* [FEATURE] Add experimental Prometheus 3 support. It allows to try the new Prometheus `v3.0.0-beta.0`. #6940
```

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,6 +1,26 @@
...
+  * Feature requests in this version won't be supported till Prometheus 3 is stable but this would let users to try the beta version with Operator and report issues if any.
+* [FEATURE] Allow Prometheus Agent to update, delete, and handle changes to DaemonSet object. #6787
```
Comment (simonpasquier) [inline CHANGELOG.md:12]: after more thoughts, I'm not sure that we need to include this in the changelog.

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,6 +1,27 @@
...
+* [FEATURE] Add support for `PodMonitor` to the `PrometheusAgent` CRD in DaemonSet mode. #6879
+* [FEATURE] Add `-kubelet-endpointslice` argument to support `EndpointSlice` for the kubelet controller. #6882
+* [FEATURE] Add `outOfOrderTimeWindow` field to the `PrometheusAgent` CRD. #6851
+* [FEATURE] Add `matchFirstNetwork` field to the `ScrapeConfig` CRD for the Docker Service Discovery. #6849 #6894
+* [FEATURE] Add the `otlp` field to the `Prometheus` and `PrometheusAgent` CRDs. #6780
+* [FEATURE] Add support for certificate, key and client CA file paths to the web TLS configuration. #6358
+* [FEATURE] Add `noProxy`, `proxyFromEnvironment` and `proxyConnectHeader` fields to the AlertmanagerConfig CRD for OAuth2 clients. #6922
+* [FEATURE] Add `noProxy`, `proxyFromEnvironment` and `proxyConnectHeader` fields to the AlertmanagerConfig CRD for HTTP clients. #6864
+* [FEATURE] Add Ionos Service Discovery support to `ScrapeConfig` CRD. (#6732)
+* [BUGFIX] Fix panic when processing an invalid `AlertmanagerConfig` object used for global configuration. #6931
```
Comment (simonpasquier) [inline CHANGELOG.md:23]: can you remove L32?

Comment (slashpai) [reply CHANGELOG.md:23]: done

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,14 +1,34 @@
 ## Unreleased
 
-* [FEATURE] Add `-kubelet-endpointslice` argument to support `EndpointSlice` for the kubelet controller.
+## 0.77.0 / 2024-09-19
+
+> Along with this release we also introduce a new command line tool [poctl](https://github.com/prometheus-operator/poctl). It is designed specifically for managing Prometheus Operator Custom Resources.
```
Comment (nicolastakashi) [inline CHANGELOG.md:5]: Nothing required, but would be nice if we callouts: https://github.com/orgs/community/discussions/16925

info and warning wdyt?

Comment (slashpai) [reply CHANGELOG.md:5]: This is nice. I had thought about updating our README as well when I saw something similar in metrics-server repo.

I will update. Thanks for the suggestion 🙂

Comment (slashpai) [reply CHANGELOG.md:6]: https://github.com/prometheus-operator/prometheus-operator/blob/0a82836229afad0c1044dfcc0d03d3c8fb13f15f/CHANGELOG.md looks nice :D

**Comment by slashpai** [comment]
would wait if we need to include https://github.com/prometheus-operator/prometheus-operator/pull/6585 as looks complete?

cc: @simonpasquier

**Comment by slashpai** [comment]
I will proceed with the release as we are one day late already, can do patch release with fix if needed later
can I get approval if changes are ok?

@simonpasquier @nicolastakashi @ArthurSens

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**